hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -494,9 +505,38 @@ private Pair<SchemaProvider, Pair<String, JavaRDD<HoodieRecord>>> fetchFromSourc
       Option<Dataset<Row>> transformed =
           dataAndCheckpoint.getBatch().map(data -> transformer.get().apply(jssc, sparkSession, data, props));
 
+      transformed = formatAdapter.processErrorEvents(transformed,
+          ErrorEvent.ErrorReason.CUSTOM_TRANSFORMER_FAILURE);
+
       checkpointStr = dataAndCheckpoint.getCheckpointForNextBatch();
       boolean reconcileSchema = props.getBoolean(DataSourceWriteOptions.RECONCILE_SCHEMA().key());
       if (this.userProvidedSchemaProvider != null && this.userProvidedSchemaProvider.getTargetSchema() != null) {
+        // If the target schema is specified through Avro schema,
+        // pass in the schema for the Row-to-Avro conversion
+        // to avoid nullability mismatch between Avro schema and Row schema
+        Option<BaseErrorTableWriter> schemaValidationErrorWriter =
+            (errorTableWriter.isPresent()
+                && props.getBoolean(HoodieErrorTableConfig.ERROR_ENABLE_VALIDATE_TARGET_SCHEMA.key(),
+                HoodieErrorTableConfig.ERROR_ENABLE_VALIDATE_TARGET_SCHEMA.defaultValue()))
+                ? errorTableWriter : Option.empty();
+        avroRDDOptional = transformed
+            .map(row ->
+                schemaValidationErrorWriter
+                    .map(impl -> {
+                      Tuple2<RDD<GenericRecord>, RDD<String>> safeCreateRDDs = HoodieSparkUtils.safeCreateRDD(row,
+                          HOODIE_RECORD_STRUCT_NAME, HOODIE_RECORD_NAMESPACE, reconcileSchema,
+                          Option.of(this.userProvidedSchemaProvider.getTargetSchema())
+                      );
+                      impl.addErrorEvents(safeCreateRDDs._2().toJavaRDD()
+                          .map(evStr -> new ErrorEvent<>(evStr,
+                              ErrorEvent.ErrorReason.AVRO_DESERIALIZATION_FAILURE)));
+                      return safeCreateRDDs._1();
+                    })
+                    .orElseGet(() -> HoodieSparkUtils.createRdd(row,
+                      HOODIE_RECORD_STRUCT_NAME, HOODIE_RECORD_NAMESPACE, reconcileSchema,
```

Comment:
lets move this to separate method and re-use in L 562 ish as well

Reply:
Addressed.